WSLBlobNFS/wsl2_linux_script.sh

```diff
@@ -60,54 +166,184 @@ function onetime_samba_setup ()
+        done
+
+        mkdir -p $mountPath
+        vecho "Created $mountPath"
+
+        mountCommand="mount -t nfs -o vers=3,proto=tcp $mountparameter $mountPath"
+        shareName="nfsv3share-$randomnumber"
+    fi
+
+    # Check if the mount path is already mounted
+    mount | grep -q $mountPath
+    if [[ $? == 0 ]]; then
+        eecho "Mount path $mountPath is already mounted. Use a different mount path."
+        exit 1
+    fi
+
+    vecho "Mounting NFS share.."
+
+    # quote the mount command to preserve the spaces
+    mount_nfs "$mountCommand" "$mountPath"
+
+    if [[ $? != 0 ]]; then
+        eecho "Failed to mount NFS share with: $mountCommand"
+        exit 1
+    fi
+
+    vecho "Done NFS mounting."
+
+    vecho "Exporting NFS share via Samba.."
+    share_via_samba "$shareName" "$mountPath"
+
+    if [[ $? != 0 ]]; then
+        eecho "Failed to export NFS share via Samba with: $shareName, $mountPath"
+
+        # Unmount the NFS share
+        unmount_nfs $mountPath
+        exit 1
+    fi
+
+    echo "$shareName" > $tempFilePath
+    vecho "Saved the share name ($sharename) to $tempFilePath"
+}
+
+# Unmount NFS
+# $1: mount path
+function unmount_nfs ()
+{
+    if [[ $# != 1 ]]; then
+        eecho "Usage: unmount_nfs <mount path>"
+        exit 1
+    fi
+
+    mntpath=$1
+    umount $mntpath
+
+    # If the umount command fails with 32, then the NFS share is already unmounted
+    if [[ $? != 0 && $? != 32 ]]; then
+        eecho "Failed to unmount NFS share at: $mntpath"
+        return 1
+    fi
+
+    vecho "Unmounted NFS mount at: $mntpath"
+    return 0
 }
 
 # Unmount smb share and NFS share
-# $1: export name
+# $1: share name
 # To-do:
 # 1. Take a lock on the smb file
 function unmount_share ()
 {
+    smbsharename=$1
     mntpath=""
     foundshare=false
     startline=0
     endline=0
     linenum=0
 
+    # Take a backup of smb.conf to restore it if the script fails
+    cp /etc/samba/smb.conf /etc/samba/smb.conf.bak
+
     # Take the last matching share name
     while IFS= read -r line; do
         # line number counter
         linenum=$((linenum + 1))
 
         # exact match of the share name
-        if [[ "$line" == "[$1]" ]]; then
+        if [[ "$line" == "[$smbsharename]" ]]; then
             foundshare=true
             startline=$linenum
 
         # next share name after the target share
-        elif [[ $foundshare == "true" && "$line" == "["*"]" ]]; then
+        elif [[ $foundshare == "true" && "$line" =~ \[.*\] ]]; then
             foundshare=false
             # Current line belongs to a new share. So, the previous line is the end of the share
             endline=$((linenum - 1))
         fi
 
         # If the path is found, get the path of the share and remove the share details from smb.conf
         # Sample line: path = /mnt/nfs/share1
-        if [[ $foundshare == "true" && "$line" == "path"* ]]; then
+        if [[ $foundshare == "true" && "$line" =~ path.** ]]; then
```

Comment:
path.** is not accurate
what you should do is (this is exact code).. dont put quotes around line or match

match="path\s*=.*"
if [[ $foundshare == "true" && $line =~ $match ]]; then